data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -46,19 +46,26 @@ class SnapshotDelta(val graphs: MutableList[Snapshot],
 
   override def snapshotAtTime(instant: Instant): AttributeGraph = {
 
-    val closestGraph = graphs.reduce(returnClosestGraph(instant))
+    //Closest graph should be based on number of logs, not time.
+    // But for it to be feasible each graph should have a sequence ID as well
+    // This is possible, I think, but not implemented
+    val closestGraph = graphs.reduceLeft((snap1, snap2) =>
+      if (Duration.between(snap2.instant, instant).isNegative) { // This means the next snapshot in the list is past the instant
```

Comment:
```suggestion
      if (snap2.instant > instant) { 
```
Er dette det samme? I så fall blir kommentaren mindre nødvendig også

Reply:
ja, det er mye bedre faktisk. Ble litt låst i hodet mitt til å bruke Duration.between